### ISSUE TITLE
Allow `--key <column name(s)>` in addition to `--key <column index(es)>` for `qsv diff`

### ISSUE THREAD

**Issue by mfripp** [jqnatividad/qsv#1999]
**Is your feature request related to a problem? Please describe.**
It is sometimes easier and usually less error-prone to identify a column by its name (entry in header row) instead of its integer index among the columns. So it would be great to allow specification of the key column(s) by name instead of just by index when running `qsv diff`. This would be particularly helpful if the location of the key columns may be different between the two files (there is currently no way to match records using key columns that have the same name but different indexes).

**Describe the solution you'd like**
Allow `qsv diff --key <column name(s)>`, e.g., `qsv diff --key type,age` or `qsv diff --key "sales rep","sales year"` when both tables have header rows. Then use the specified columns for matching when comparing rows.

**Describe alternatives you've considered**

`--key-name <column name(s)>` would be possible, but it doesn't seem necessary to add a different flag for this

Currently I just make do by counting the index of the columns and making sure it is the same in both tables.

**Additional context**
(none)

**Comment by jqnatividad** [comment]
I agree that it'd be convenient to support column names with the `--key` option and can certainly add it.

However, the `--key` column names still need to be in the same place for both left and right CSVs, as that is a requirement of the underlying library - `csv-diff`, that we're using.

That is, if we add column name support, both CSVs will have to have the selected names in the same index locations.

Perhaps, use `qsv select` first to reorder the columns before `diff`ing?

**Comment by jqnatividad** [comment]
@mfripp , I ended up implementing your request with the limitation that the column names are in the same index location for both left and right CSVs.

It's more of a convenience shortcut, so if the CSV headers are ordered differently, the workaround to reorder the column using `select` before `diff`ing remains.

I also added a check in there to detect if the colnames do not match and provide actionable info about the `select` workaround.
